File: Go/control.go
package main

import (
	. "fmt"
)

func main() {

	a := 5
	if a < 0 {
		Println("负数")  // 即使只有一条语句，也必须有花括号
	} else if a > 0 {
		Println("正数")
	} else {
		Println(0)
	}



	if b := a - 1; b > 0 {
		Println(b)  // b只在if内有效，出了if就释放了
	}



	c := 1
	switch c + 1 {
		case 1:
			Println("A")
		case 2, 3, 4:
			Println("B")
		default:
			Println("C")
	}



	for i := 0; i < 10; i++ {
		Println(i)  // 同上，i出了for就无效了
		if i == 5 {
			break;  // 也支持continue
		}
	}



	// 相当于where
	i := 0
	for i < 10 {
		Println(i)
		i++
	}



	i = 0
	Label:
		Println(i)
		i++
		if( i != 5 ) {
			goto Label  // 只能跳到当前函数内定义的标签
		}
}

File: 编程语言 快速启动/PHP/Control.php
<?php

    $var = 1;

    if ( $var < 10 ) {
        /* ... */
    } else if ( $var < 20 ) {
        /* ... */
    } else {
        /* ... */
    }



    /* $var会自动转换为case对应的类型再比较 */
    switch ( $var ) {
        case 1:
            /* ... */ break;
        case 'string':
            /* ... */ break;
        case false:
            /* ... */ break;
        case array( 1, 2, 3 ):
            /* ... */ break;
        default:
            /* ... */
    }



    for ( $i = 0; $i < 10; $i++ ) {
        /* ... */
    }
    // $i用完后仍存在，这里为10



    $i = 0;
    while ( $i < 10 ) {
        /* ... */;
        $i++;
    }



    $i = 0;
    do {
        /* ... */
        $i++;
    } while ( $i < 10 );

    /* for, while, do-while中皆可使用continue与break */



    goto Label;
    echo 1;  // 跳过了
    Label:
        echo 2;

    /*
     * goto可以往前、往后跳；
     * 可以跳出switch或循环，但无法跳入
     */

?>
